src/Illuminate/Database/Console/Factories/FactoryMakeCommand.php

```diff
@@ -46,11 +47,18 @@ protected function getStub()
      */
     protected function buildClass($name)
     {
-        $namespaceModel = $this->option('model')
-                        ? $this->qualifyClass($this->option('model'))
-                        : trim($this->rootNamespace(), '\\').'\\Model';
+        $model = class_basename($this->option('model') ?? $name);
 
-        $model = class_basename($namespaceModel);
+        foreach (['factory', 'Factory'] as $name) {
+            if (Str::endsWith($model, $name) && strpos($model, $name)) {
+                $model = substr($model, 0, 4);
```

Comment:
What is this doing? Why are you slicing off a hard-coded 4 characters?

Reply:
Oh sorry about that . Was thinking of slicing the last 7 characters of `Factory` but didn't notice the `4` it is fixed now on the last commit.

Reply:
@simoebenhida Might be nice to tuck it behind a nicely-named method, i.e.

```php
$model = $this->removeFactorySuffix($model);
```